Kudu.Contracts/SiteExtensions/ISiteExtensionManager.cs

```diff
@@ -1,20 +1,20 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Kudu.Contracts.SiteExtensions
 {
     public interface ISiteExtensionManager
     {
-        IEnumerable<SiteExtensionInfo> GetRemoteExtensions(string filter, bool allowPrereleaseVersions, string feedUrl);
+        Task<IEnumerable<SiteExtensionInfo>> GetRemoteExtensions(string filter, bool allowPrereleaseVersions, string feedUrl);
 
-        SiteExtensionInfo GetRemoteExtension(string id, string version, string feedUrl);
+        Task<SiteExtensionInfo> GetRemoteExtension(string id, string version, string feedUrl);
 
-        IEnumerable<SiteExtensionInfo> GetLocalExtensions(string filter, bool checkLatest);
+        Task<IEnumerable<SiteExtensionInfo>> GetLocalExtensions(string filter, bool checkLatest);
 
-        SiteExtensionInfo GetLocalExtension(string id, bool checkLatest);
+        Task<SiteExtensionInfo> GetLocalExtension(string id, bool checkLatest);
 
-        SiteExtensionInfo InstallExtension(string id, string version, string feedUrl);
+        Task<SiteExtensionInfo> InstallExtension(string id, string version, string feedUrl);
```

Comment:
Yes, we should talk about this. We also have one major issue today, which is that if you install via ARM (or direct Kudu), you end up with your new extension not being active until you restart the site. This really screws up ARM templates (like [this one](https://github.com/davidebbo/AzureWebsitesSamples/blob/master/ARMTemplates/WebSiteWithSiteExtension.json)) that try to install an extension and then call into it.